src/mds/MDSRank.cc

```diff
@@ -3054,11 +3054,77 @@ bool MDSRankDispatcher::handle_command(
 
     damage_table.erase(id);
     return true;
+  } else if (prefix == "drop cache") {
+    command_drop_cache(ds);
+    return true;
   } else {
     return false;
   }
 }
 
+void MDSRank::command_drop_cache(stringstream *ds)
+{
+  if (not is_active()) {
+    dout(5) << __func__ << ": MDS not active, no-op" << dendl;
+    return;
+  }
+
+  if (not mdcache->trim(UINT64_MAX))
+    return;
```

Comment:
I think trimming the cache should occur after we get all the client recall acks. We want to trim the MDS cache when the client caps are released.